routers/web/repo/issue_lock.go

```diff
@@ -21,13 +21,13 @@ func LockIssue(ctx *context.Context) {
 
 	if issue.IsLocked {
 		ctx.Flash.Error(ctx.Tr("repo.issues.lock_duplicate"))
-		ctx.Redirect(issue.Link())
+		ctx.JSONRedirect(issue.Link())
```

Comment:
Remove `Flash.Error` , use `JSONError`

Reply:
Done in https://github.com/go-gitea/gitea/pull/25380/commits/901a2dc8f2dcaac67df165d3b566170f23c45712 and tested with incorrect lock type

<img width="516" alt="Screen Shot 2023-06-20 at 13 18 11" src="https://github.com/go-gitea/gitea/assets/17645053/daf12da0-41b4-43c9-a2ce-cc1c647876d9">